pyproject.toml

```diff
@@ -23,30 +23,29 @@ repository = "https://github.com/prql/pyprql"
 version = "0.12.1"
 
 [tool.poetry.dependencies]
-python = "^3.8"
+python = "^3.10"
```

Comment:
Bumped this all the way to 3.10 since otherwise requires careful management of ipython dependency

Colab is on 3.10, which is some standard

Reply:
Thanks for working on this.

> Bumped this all the way to 3.10 since otherwise requires careful management of ipython dependency
> 
> Colab is on 3.10, which is some standard

Hmmm, I don't know if it's worth dropping Python 3.9 just for that.

Isn't Colab based on 3.10 because python3 on Ubuntu 22.04 is 3.10?
<https://launchpad.net/ubuntu/jammy/+package/python3>

That has nothing to do with the release cycle of Python itself (since Ubuntu LTS is released every two years)

Reply:
I don't think it's related to Ubuntu: https://colab.google/articles/py3.10

Reply:
Thanks for the context!

In any case, I don't know if there's any justification for dropping Python 3.9, given that duckdb, pyarrow and polars support 3.9.

Reply:
I agree it's not ideal. But pandas etc have dropped 3.9, and it seems like we're going to need to do some dependency surgery to make ipython work with it — I'm mostly thinking whether that's worthwhile...

Reply:
I don't understand why the pandas version is relevant here.
pandas does not seem a dependency of IPython.

Reply:
Yes, I just mean that `pandas` is a big library that others follow — it's not like every other library support 3.9 except for us. Does that make more sense?